ui/app/AppLayouts/Profile/views/PasswordView.qml

```diff
@@ -0,0 +1,356 @@
+        if(newPswInput.text.length >= root.minPswLen) {
+            errorTxt.text = ""
+            if(confirmPswInput.text !== newPswInput.text) {
+                errorTxt.text = qsTr("Passwords don't match")
+            }
+        }
+    }
+
+    QtObject {
+        id: d
+
+        property bool containsLower: false
+        property bool containsUpper: false
+        property bool containsNumbers: false
+        property bool containsSymbols: false
+
+        readonly property var validator: RegExpValidator { regExp: /^[!-~]+$/ } // That incudes NOT extended ASCII printable characters less space
+
+        // Password strength categorization / validation
+        function lowerCaseValidator(text) { return (/[a-z]/.test(text)) }
+        function upperCaseValidator(text) { return (/[A-Z]/.test(text)) }
+        function numbersValidator(text) { return (/\d/.test(text)) }
+        // That incudes NOT extended ASCII printable symbols less space:
+        function symbolsValidator(text) { return (/[!-\/:-@[-`{-~]/.test(text)) }
+        function findUniqueChars(text) {
+            // The variable that contains the unique values
+            let uniq = "";
+
+            for(let i = 0; i < text.length; i++) {
+                // Checking if the uniq contains the character
+                if(uniq.includes(text[i]) === false) {
+                    // If the character not present in uniq
+                    // Concatenate the character with uniq
+                    uniq += text[i]
+                }
+            }
+            return uniq
+        }
+
+        // Algorithm defined in functional requirements / Password categorization
+        function getPswStrength() {
+            let rules = 0
+            let points = 0
+            let strengthType = StatusPasswordStrengthIndicator.Strength.None
+
+            if(newPswInput.text.length >= root.minPswLen) { points += 10; rules++ }
+            if(d.containsLower) { points += 5; rules++ }
+            if(d.containsUpper) { points += 5; rules++ }
+            if(d.containsNumbers) { points += 5; rules++ }
+            if(d.containsSymbols) { points += 10; rules++ }
+
+            let uniq = d.findUniqueChars(newPswInput.text)
+            if(uniq.length >= 5) { points += 5; rules++ }
+
+            // Update points according to rules used:
+            points += rules * 10/*factor*/
+
+            // Strength decision taken:
+            if(points > 0 && points < 40) strengthType = StatusPasswordStrengthIndicator.Strength.VeryWeak
+            else if(points >= 40 && points < 60) strengthType = StatusPasswordStrengthIndicator.Strength.Weak
+            else if(points >= 60 && points < 80) strengthType = StatusPasswordStrengthIndicator.Strength.SoSo
+            else if(points >= 80 && points < 100) strengthType = StatusPasswordStrengthIndicator.Strength.Good
+            else if(points >= 100) strengthType = StatusPasswordStrengthIndicator.Strength.Great
+            return strengthType
```

Comment:
You can clean the `if`s here by just returning directly in teh `if`. eg:
```qml
if(points > 0 && points < 40) return StatusPasswordStrengthIndicator.Strength.VeryWeak
```
And at the end, just return `StatusPasswordStrengthIndicator.Strength.None`. The `strengthType` is then not needed.

Reply:
I understand your suggestion but I like to concentrate just only one return point in a function, whenever possible, to improve the function readability because sometimes if the function is too long, you could be lost trying to know which is at the end the result depending on the condition. It sometimes requires to write more code lines but I think it should be a good balance..

In that particular case, yes, we could use your approach as all the return points are concetrated at the end and will not penalize the readability. But, have I convinced you to keep it as it is with my argument :yum: ?

Reply:
>  But, have I convinced you to keep it as it is with my argument 😋 ?

Haha, I won't force you to change the code, since it's a minor change. I still prefer my way of doing, but I know you're not the only one that doesn't like early returns, so to each their own!